src/wireviz/wv_colors.py

```diff
@@ -104,15 +104,18 @@
 def get_color_hex(input, pad=False):
     if input is None or input == '':
         return [color_default]
+
     if len(input) == 4:  # give wires with EXACTLY 2 colors that striped/banded look
-        input = input + input[:2]
-    # hacky style fix: give single color wires a triple-up so that wires are the same size
-    if pad and len(input) == 2:
-        input = input + input + input
+        padded = input + input[:2]
+    elif pad and len(input) == 2: # hacky style fix: give single color wires a triple-up so that wires are the same size
+        padded = input + input + input
+    else:
+        padded = input
+
     try:
-        output = [_color_hex[input[i:i + 2]] for i in range(0, len(input), 2)]
+        output = [_color_hex[padded[i:i + 2]] for i in range(0, len(input), 2)]
```

Comment:
- Test: YAML input from https://github.com/formatc1702/WireViz/issues/63#issuecomment-668757989
- Bug: The returned output list length is now `len(input) / 2`, i.e. ignoring any extra entries in `padded`
- Fix: Replace `len(input)` with `len(padded)`